spec.bs

```diff
@@ -1586,6 +1589,25 @@ an [=interest group=] |ig|, a [=boolean=] |isComponentAuction| and a [=boolean=]
     [=InterestGroupReportingScriptRunnerGlobalScope/report url=] to |parsedUrl|.
 </div>
 
+<div algorithm>
+  The <dfn method for="InterestGroupReportingScriptRunnerGlobalScope">registerAdBeacon(|map|)</dfn>

```

Comment:
Do the two instances of throwing an exception here match what the implementation does? It would be nice to if this were backed by WPTs, but at the very least a spot check here would be good.

Reply:
Hmm shouldn't the map always be present? if so we should make a few changes:
1. [=evalute a reporting script=] should always set [=InterestGroupReportingScriptRunnerGlobalScope/reporting beacon map=] to a new empty map or an existing map
2. We should [=Assert=] that it's not null below instead of throwing
3. We shouldn't say "or null" when we define [=InterestGroupReportingScriptRunnerGlobalScope/reporting beacon map=] above?

Reply:
> Do the two instances of throwing an exception here match what the implementation does? It would be nice to if this were backed by WPTs, but at the very least a spot check here would be good.

Yes, I wrote this following the implementation.

Reply:
> Hmm shouldn't the map always be present?
registerAdBeacon throws if you call it more than once. Having the map initially be null allows us to implement this check very easily.

When it gets copied into the fenced frame config in a later PR, I will just have to handle `null` appropriately.